graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/MutationFetcher.java

```diff
@@ -74,9 +77,33 @@ protected MutationFetcher(
     }
 
     // Execute as a single statement
-    return dataStore
-        .execute(query)
-        .thenApply(rs -> ImmutableMap.of("value", environment.getArgument("value")));
+    return dataStore.execute(query).thenApply(rs -> toMutationResult(rs, environment));
+  }
+
+  private Map<String, Object> toMutationResult(
+      ResultSet resultSet, DataFetchingEnvironment environment) {
+
+    List<Row> rows = resultSet.currentPageRows();
+    Object originalValue = environment.getArgument("value");
+
+    if (rows.isEmpty()) {
+      // if we have no rows means that we got no information back from query execution, return
+      // original value to the user
+      return ImmutableMap.of("value", originalValue);
+    } else {
+      // otherwise check what can we get from the results
+      // mutation target only one row
+      Row row = rows.iterator().next();
+      boolean applied = row.getBoolean("[applied]");
+      Map<String, Object> value = DataTypeMapping.toGraphQLValue(nameMapping, table, row);
+
+      // return value from query, if given or not applied, fallback to the original
+      Object finalValue = !value.isEmpty() || !applied ? value : originalValue;
```

Comment:
I wonder if we should consider `value.isEmpty()` in this condition.

The only cases when `!applied && value.isEmpty()` are failed `IF EXISTS` queries:
```
cqlsh:test> delete from foo where k = 1 if exists;

 [applied]
-----------
     False
```
With the current code we'll return the input data, which is not consistent with CQL (and to be clear this was already the case before, it's not a consequence of your changes).

I think the idea is when `applied == false` is to reflect the conflicting database state that caused the LWT to fail. If that state is "there was nothing to delete", then returning an empty value (like cqlsh does) is the right thing to do.

So I would simply do:
```suggestion
      Object finalValue = applied ? originalValue : value;
```

This is a breaking change, but I think the previous behavior was wrong.
WDYT?

Reply:
This is more than valid imo..